### PR TITLE
[protobuf] fix dynamic build

### PR DESCRIPTION
**Pull request by Jamlys**
**Describe the pull request**

- #### What does your PR fix?
  Fix failure of dynamic build.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  all, No

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Basically

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/protobuf/fix-vcpkg-dynamic-build.patch [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+     RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT protoc
+     BUNDLE DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT protoc)
+-  if (UNIX AND NOT APPLE)
++  if (UNIX AND NOT APPLE AND NOT protobuf_MSVC_STATIC_RUNTIME)
+     set_property(TARGET protoc
+-      PROPERTY INSTALL_RPATH "$ORIGIN/../${CMAKE_INSTALL_LIBDIR}")
++      PROPERTY INSTALL_RPATH "$ORIGIN:$ORIGIN/../${CMAKE_INSTALL_LIBDIR}:$ORIGIN/../../${CMAKE_INSTALL_LIBDIR}")
+   elseif (APPLE)
+     set_property(TARGET protoc
+       PROPERTY INSTALL_RPATH "@loader_path/../lib")
```
Comment (cenit) [inline ports/protobuf/fix-vcpkg-dynamic-build.patch:25]: only this part of patch file should survive, everything else around is an unnecessary waste of storage space in the long term

Comment (Jamlys) [reply ports/protobuf/fix-vcpkg-dynamic-build.patch]: thx for suggestion. :)
